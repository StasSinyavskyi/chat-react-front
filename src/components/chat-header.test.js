/* eslint-env jest */
import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import ChatHeader from './chat-header';

jest.mock('./chat-action-menu', () => () => 'ChatActionMenu');
jest.mock('../containers/chat-header-menu', () => () => 'ChatHeaderMenu');
jest.mock('./avatar', () => () => 'Avatar');

// if we have several test we can join them in describe
describe('<ChatHeader />', () => {
  const testParam = (
    <ChatHeader
      liveChat={() => 'liveChat'}
      deleteChat={() => 'deleteChat'}
      activeUser={{ }}
      activeChat={{ }}
      disabled={false}
      isConnected={false}
    />
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(testParam, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render correctly', () => {
    const tree = renderer
      .create(testParam)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
