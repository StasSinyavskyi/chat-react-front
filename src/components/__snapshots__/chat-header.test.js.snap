// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatHeader /> render correctly 1`] = `
<header
  className="MuiPaper-root-82 MuiPaper-shadow4-88 MuiAppBar-root-74 MuiAppBar-positionFixed-75 MuiAppBar-colorPrimary-80 mui-fixed Chatheader-appBar-68"
>
  <div
    className="MuiToolbar-root-109 MuiToolbar-gutters-110 Chatheader-appToolBar-70"
  >
    Avatar
    <h2
      className="MuiTypography-root-111 MuiTypography-title-117 MuiTypography-colorInherit-130 Chatheader-appBarTitle-71"
    />
    ChatActionMenu
  </div>
  ChatHeaderMenu
</header>
`;
